### PR TITLE
[Core] Include default gherkin version in version.properties

### PR DESCRIPTION
**Pull request by davidjgoss**
## Summary

Expose the gherkin version in `version.properties`, so it can be picked up from the classpath by other code.

## Motivation and Context

A bit more context in https://github.com/cucumber/cucumber/issues/779#issuecomment-561220566, but I am specifically targeting the IDEA autocompletion support for Rule/Example grammar.

I am opening this now for more of a steer, as I can see that with #1804 and others we now have multiple Gherkin versions so this would no longer be valid, and I'm not sure what the best way to achieve this kind of feature detection would be now - or if it's the right thing to do at all.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

(Sorry, meant to open this as a draft)

### CODE REVIEW

**Comment by mpkorstanje** [comment]
Mmh. This is an interesting problem.

**Comment by mpkorstanje** [comment]
While the introduction of the compatibility layer may suggest otherwise, the plan for Cucumber is not to support multiple versions of Gherkin indefinitely. Rather we are trying to solve a migration problem.

Cucumbers `json` and `html` plugins create reports on the execution of Cucumber and they do so by creating a facsimile of the original feature file. As a result they are quite complex and hard to maintain. And on top of that they are duplicated in the different implementations of Cucumber. Finally the output produced by the `json` formatter doesn't have a schema and differs per cucumber implementation.

We're looking to solve several problems all at once by introducing a new output format based on protobuf/ndjson. However as there are no consumers of protobuf/ndjson yet we can't simply remove the `json` formatter. We'll have to support it for at least a little while. This means that internally we need parsers for both Gherkin 5 and Gherkin 8+.

File: core/src/main/resources/io/cucumber/core/version.properties [hunk 1]
```diff
@@ -1 +1,2 @@
-cucumber-jvm.version=${parent.version}
\ No newline at end of file
+cucumber-jvm.version=${parent.version}
+gherkin.version=${gherkin-messages.version}
```
Comment (mpkorstanje) [inline core/src/main/resources/io/cucumber/core/version.properties]: Currently (RC3) we support Gherkin 5 by default and Gherkin 8+ (rule keyword) as an opt in module. However only JUnit 5 can accurately represent these outputs so I wouldn't enable it by default. Once we've finished the protobuf/ndjson output we'll probably enable Gherkin 8+ by default. 

Which is a really round about way to say that I think this should be: 
```suggestion
gherkin.version=${gherkin-vintage.version}
```

**Comment by davidjgoss** [comment]
@mpkorstanje okay yeah - so we declare the vintage version for now as it's the default. Thanks!

**Pull request merged**